activemodel/test/cases/helper.rb

```diff
@@ -16,6 +16,16 @@
 class ActiveModel::TestCase < ActiveSupport::TestCase
   include ActiveSupport::Testing::MethodCallAssertions
 
+  class AssertionlessTest < StandardError; end
+
+  def after_teardown
+    super
+    return if !passed? || assertions > 0
+
+    message = "#{self.name} performs no assertions."
+    raise AssertionlessTest, message
```

Comment:
Could thoughts:

- I think `self.` is redundant here, but also presumably the reporter already includes the name of the failing test, so I don't think we need to repeat it, especially since test names are often long.
- Semantic nit: an assertion is "made", not "performed"
- Something about `return` as the guard clause feels weird here.

What do you think about this?


```suggestion
    raise AssertionlessTest, "No assertions made." if passed? && assertions.zero?
```

Or alternatively, could we reuse the assertion machinery and assert on assertions, rather than use a custom exception?

```suggestion
assert_not(passed? && assertions.zero?, "No assertions made")
```

Reply:
I like it! I inlined everything as you suggested but decided to stick to `raise` for now but I'll think a bit more about using an assertion failure instead.

The difference between raising and using an assertion is how the failure is going to be reported at the end. Raised exception will be reported as an `error` while failed assertion will be reported as a `failure`. It's not a big deal but if someone pays attention to the difference it may confuse them and make an impression like the test is failing because the application logic is wrong while I'd prefer to make it as clear as possible that the problem is with the test itself and not with the logic the test covers